pkg/plugins/approve/approvers/owners.go

```diff
@@ -73,6 +74,19 @@ func (o Owners) GetApprovers() map[string]sets.String {
 	return ownersToApprovers
 }
 
+// GetRequiredApproversCount returns the amount of approvers required to get a PR approved.
```

Comment:
This definition of GetRequiredApproversCount is not sufficient since it doesn't check whether the approvers are the correct ones for the files. Actually this function and the use of it should be replaced with a check in `UnapprovedFiles` where you should replace

`		if len(approvers) == 0 {
`

with 

`		if len(approvers) < ap.owners.repo.MinimumReviewersForFile(fn) {
`

Or some other equivalent change. 

GetFiles should get the equivalent and then using this TestGetFiles should get test cases for minimum_reviewers.

Reply:
@msvticket Been having difficulties implementing this as `GetOwnersSet()` (https://github.com/spring-financial-group/lighthouse/blob/main/pkg/plugins/approve/approvers/owners.go#L176) doesn't seem to return the OWNERS files that I'm expecting.

`o.removeSubdirs()` means that the highest level OWNERS file will always be returned for a given change, although I've now noticed that the `NoParentOwners` field allows you to override this. To me `NoParentOwners` should be the default behaviour (in fact this is how I thought it worked 😂), I'm wondering when you would ever not want this to be the case?

Reply:
Now I'm aware of `NoParentOwners` I should be able to implement, this functionality just confused me 😂